### ISSUE TITLE
Java language server fails to load, jdt process is hung

### ISSUE THREAD

**Issue by jherico** [redhat-developer/vscode-java#2047]
This is superficially like #158 but I believe this is actually a new issue caused by a very recent change.  

That bug has a recent comment linking to [this](https://github.com/redhat-developer/vscode-java/issues/2020#issuecomment-880272712) comment on #2020 which in turn links to [this](https://bugs.eclipse.org/bugs/show_bug.cgi?id=574227) very recently fixed bug in Eclipse JDT-APT.  The fix appears to have wrecked my ability to load my maven based projects.  

Specifically, when I load a Maven project, eventually all CPU usage in the Java child processes ceases and when I run jstack against the JDT Java process I see this frame

```
"Worker-1: Initialize Workspace" #30 prio=5 os_prio=0 cpu=2996.36ms elapsed=39.67s tid=0x00007ff6bc001800 nid=0x5362 in Object.wait()  [0x00007ff71596c000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(java.base@11.0.11/Native Method)
        - waiting on <no object reference available>
        at org.eclipse.core.internal.jobs.JobManager.join(JobManager.java:1020)
        - waiting to re-lock in wait() <0x0000000082f00000> (a java.util.Collections$SynchronizedSet)
        at org.eclipse.jdt.apt.core.internal.generatedfile.GeneratedSourceFolderManager.removeFolder(GeneratedSourceFolderManager.java:508)
        at org.eclipse.jdt.apt.core.internal.generatedfile.GeneratedSourceFolderManager.folderNamePreferenceChanged(GeneratedSourceFolderManager.java:321)
        at org.eclipse.jdt.apt.core.internal.AptProject.preferenceChanged(AptProject.java:81)
        at org.eclipse.jdt.apt.core.util.AptConfig.setString(AptConfig.java:969)
        at org.eclipse.jdt.apt.core.util.AptConfig.setGenSrcDir(AptConfig.java:917)
```

The `GeneratedSourceFolderManager.removeFolder()` function at line 508 in the 0.80.0 version of this plugin is the newly introduced code by the the fix commit [here](https://git.eclipse.org/r/c/jdt/eclipse.jdt.core/+/182187).  It looks as if the code is suffering from some kind of race condition and waiting on something that never occurs, causing the language server to eventually timeout.

As soon as I rolled back to 0.79.2 I'm able to load the project.

##### Environment
- Operating System: Ubuntu 20.04.2 LTS
- JDK version: OpenJDK Runtime Environment (build 11.0.11+9-Ubuntu-0ubuntu2.20.04)
- Visual Studio Code version: 1.58.2
- Java extension version: 0.80.0

##### Steps To Reproduce
1. Open a maven multi-module project

I can't attach an actual project since it's not open source, but I can attach the pom files.

[parent pom](https://gist.github.com/jherico/b8f21f7ba1f903841662e77624baa453)
[common module pom](https://gist.github.com/jherico/cf36def99f83709e978ef03f5e2c26cc)
[plans module pom](https://gist.github.com/jherico/7e5308c55e9167e606b1af931a48c45e)

##### Current Result

Java projects window never populates.  

##### Expected Result

Java projects window populates with all the modules

**Comment by snjeza** [comment]
@jherico could you check https://github.com/redhat-developer/vscode-java/issues/2020#issuecomment-880272712
